kalite/distributed/management/commands/screenshots.py

```diff
@@ -0,0 +1,338 @@
+        # since we used a new sqlite database for this command we must delete it after
+        database_name = getattr(settings, 'SCREENSHOTS_DATABASE_NAME', None)
+        if database_name:
+            delete_sqlite_database(sc.database, verbosity=options['verbosity'])
+
+
+# Possibly re-usable codes.
+def reset_sqlite_database(username=None, email=None, password=None, router=None, verbosity="1"):
+    """
+    Resets the currently used sqlite database.  Creates the user if admin_username is passed.
+    :param username: If present, creates a superuser with this username.
+    :param email: If present, creates a superuser with this email.
+    :param password: If present, creates a superuser with this password.
+    :param router: The database router to use.
+    :return: Returns the superuser created or None if no arguments are provided.
+    """
+    if not router:
+        router = getattr(settings, 'SCREENSHOTS_ROUTER', 'default')
+    db_engine = settings.DATABASES[router]['ENGINE']
+    if db_engine == settings.SQLITE3_ENGINE:
+
+        # make sure database path exists
+        ensure_dir(settings.SCREENSHOTS_OUTPUT_PATH)
+
+        new_io = StringIO()
+        call_command("reset_db", interactive=False, stdout=new_io, router=router, verbosity=verbosity)
+        call_command("syncdb", interactive=False, stdout=new_io, router=router, verbosity=verbosity)
+        call_command("migrate", interactive=False, stdout=new_io, router=router, verbosity=verbosity)
+        if username and email and password:
+            log.info('==> Creating superuser username==%s; email==%s ...' % (username, email,)) if int(verbosity) > 0 else None
+            call_command("createsuperuser", username=username, email=email,
+                         interactive=False, stdout=new_io, router=router, verbosity=verbosity)
+            admin_user = User.objects.get(username=username)
+            admin_user.set_password(password)
+            admin_user.save()
+            return admin_user
+    return None
+
+
+def delete_sqlite_database(database=None, verbosity="1"):
+    """
+    Delete the specified sqlite database or if None, the one on `settings.py` of the app or project.
+    :param database: The database filename with full path.
+    """
+    try:
+        router = getattr(settings, 'SCREENSHOTS_ROUTER', 'default')
+        db_engine = settings.DATABASES[router]['ENGINE']
+        if db_engine == settings.SQLITE3_ENGINE:
+            if not database:
+                database = settings.DATABASES[router]['NAME']
+            if os.path.exists(database):
+                log.info('==> Removing database %s ...' % database) if int(verbosity) > 0 else None
+                os.remove(database)
+                log.info('====> Successfully removed database.') if int(verbosity) > 0 else None
+    except Exception as exc:
+        log.error('====> EXCEPTION: %s' % exc)
+
+
+class Screenshot(KALiteBrowserTestCase, FacilityMixins, BrowserActionMixins):
```

Comment:
Noted... perhaps we should also make a note of it in the mixins files? Not obvious to me that the inheritance is sensitive to order.